packages/event-handler/tests/unit/BedrockAgentFunctionResolver.test.ts

```diff
@@ -0,0 +1,457 @@
+      {
+        name: 'a',
+        type: 'number',
+        value: '1',
+      },
+      {
+        name: 'b',
+        type: 'number',
+        value: '2',
+      },
+    ]);
+
+    const actual = await app.resolve(event, context);
+
+    expect(actual.response.function).toEqual('mult');
+    expect(actual.response.functionResponse.responseBody.TEXT.body).toEqual(
+      'Error: tool has not been registered in handler.'
+    );
+  });
+
+  it('overwrites tools with the same name and uses the latest definition', async () => {
+    // Prepare
+    const app = new BedrockAgentFunctionResolver();
+
+    const event = createEvent('math', [
+      {
+        name: 'a',
+        type: 'number',
+        value: '10',
+      },
+      {
+        name: 'b',
+        type: 'number',
+        value: '2',
+      },
+    ]);
+
+    app.tool(
+      async (params) => {
+        return Number(params.a) + Number(params.b);
+      },
+      {
+        name: 'math',
+        definition: 'Adds two numbers',
+      }
+    );
+
+    const addResult = await app.resolve(event, context);
+    expect(addResult.response.function).toEqual('math');
+    expect(addResult.response.functionResponse.responseBody.TEXT.body).toEqual(
+      '12'
+    );
+
+    app.tool(
+      async (params) => {
+        return Number(params.a) * Number(params.b);
+      },
+      {
+        name: 'math',
+        definition: 'Multiplies two numbers',
+      }
+    );
+
+    const multiplyResult = await app.resolve(event, context);
+    expect(multiplyResult.response.function).toEqual('math');
+    expect(
+      multiplyResult.response.functionResponse.responseBody.TEXT.body
+    ).toEqual('20');
+
+    app.tool(
+      async (params) => {
+        return Number(params.a) / Number(params.b);
+      },
+      {
+        name: 'math',
+        definition: 'Divides two numbers',
+      }
+    );
+
+    const divideResult = await app.resolve(event, context);
+    expect(divideResult.response.function).toEqual('math');
+    expect(
+      divideResult.response.functionResponse.responseBody.TEXT.body
+    ).toEqual('5');
+
+    expect(console.warn).toHaveBeenCalledTimes(3);
+    expect(console.warn).toHaveBeenCalledWith(
+      'Tool math already registered. Overwriting with new definition.'
+    );
+  });
+
+  it('can be invoked using the decorator pattern', async () => {
+    // Prepare
+    const app = new BedrockAgentFunctionResolver();
+
+    class Lambda {
+      @app.tool({ name: 'hello', definition: 'Says hello' })
+      async helloWorld() {
+        return 'Hello, world!';
+      }
+
+      @app.tool({ name: 'add', definition: 'Adds two numbers' })
+      async add(params: { a: string; b: string }) {
+        const { a, b } = params;
+        return Number.parseInt(a) + Number.parseInt(b);
+      }
+
+      public async handler(event: BedrockAgentFunctionEvent, context: Context) {
+        return app.resolve(event, context);
+      }
+    }
+
+    const lambda = new Lambda();
```

Comment:
This should have another line below like:

```
const handler = lambda.handler.bind(lambda);
```

and `handler` is what you call, which is how Lambda will call your handler.